app/components/course-page/completed-step-notice.ts

```diff
@@ -1,30 +1,55 @@
-import { service } from '@ember/service';
 import Component from '@glimmer/component';
+import type AnalyticsEventTrackerService from 'codecrafters-frontend/services/analytics-event-tracker';
 import type CoursePageStateService from 'codecrafters-frontend/services/course-page-state';
+import type RepositoryModel from 'codecrafters-frontend/models/repository';
 import type { Step } from 'codecrafters-frontend/utils/course-page-step-list';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
 
 type Signature = {
   Element: HTMLDivElement;
 
   Args: {
+    repository: RepositoryModel;
     step: Step;
   };
 };
 
 export default class CompletedStepNoticeComponent extends Component<Signature> {
+  @service declare analyticsEventTracker: AnalyticsEventTrackerService;
   @service declare coursePageState: CoursePageStateService;
 
+  @tracked shareProgressModalIsOpen = false;
+
   get activeStep() {
     return this.coursePageState.activeStep;
   }
 
+  get currentStepAsCourseStageStep() {
+    return this.coursePageState.currentStepAsCourseStageStep;
+  }
+
   get instructionsMarkdown() {
     return this.args.step.completionNoticeMessage!;
   }
 
   get nextStep() {
     return this.coursePageState.nextStep;
   }
+
+  get shouldShowShareProgressButton() {
+    return this.activeStep?.type === 'CourseStageStep' && !this.currentStepAsCourseStageStep?.stageListItem?.stage?.isFirst;
```

Comment:
@libmartinito I think you're looking for `currentStep`, not `activeStep`. Current == the page they're currently on, Active == the next stage for them to attempt

Reply:
Open to better names here btw - I can see how this is confusing